x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/latest/Latest.java

```diff
@@ -73,15 +73,24 @@ public ChangeCollector buildChangeCollector(String synchronizationField) {
         return new LatestChangeCollector(synchronizationField);
     }
 
-    private static Map<String, Object> convertBucketToDocument(CompositeAggregation.Bucket bucket,
-                                                               LatestConfig config,
-                                                               TransformIndexerStats transformIndexerStats) {
+    private static Map<String, Object> convertBucketToDocument(
+        CompositeAggregation.Bucket bucket,
+        LatestConfig config,
+        TransformIndexerStats transformIndexerStats,
+        TransformProgress progress
+    ) {
         transformIndexerStats.incrementNumDocuments(bucket.getDocCount());
+        if (progress != null) {
```

Comment:
Are we only allowing `null` for the sake of tests? If so, that seems unnecessary. Seems like we should not allow this to be nullable.

Reply:
progress is `null` for previews, however I agree with you, it would be better if we make this mandatory and fix preview.